ccswbs/gus
# Fix null path alias and exclude archived content

# Summary of changes
Fix bug from https://dev.azure.com/uoguelph/CCS/_workitems/edit/75971 and exclude archived content from the build process

## Frontend
- Refactor GraphQL query in `gatsby-node.js` to exclude:
    - content where `path.alias` is null
    - content with `moderation_state` set to archived
- Other changes in  `gatsby-node.js`:
    - Delete commented-out article code
    - Add `nodeTitle` as parameter in `processPage` function
    - Refactor `createContentTypeAlias` function for proper null check and use of `nodeTitle`
- Refactor GraphQL query in `index.js` to filter published vs draft content
- Streamline and improve other functions in `index.js`

### Incidental Changes
- Create `netlify.toml` file
- Configure Netlify to use its own image CDN to speed up build times

[ x ] My changes are accessible (at minimum WCAG 2.0 Level AA)
[ x ] My changes are responsive and appear as expected on mobile and desktop views.
[ x ] After merging, I will update the [Content Hub documentation](https://uoguelphca.sharepoint.com/sites/UniversityContentHubInformationGroup) and ensure the Content Hub trainer understands how my changes will affect users.

## Backend
N/A

# Test Plan

- Go to https://fixpath--test-ugconthub.netlify.app/
- Verify the various sections (Published vs Unpublished, Tagged vs Untagged, etc) appear as expected
- Verify the following backend page with a null alias - https://tq-jsonfix-chug.pantheonsite.io/node/3019/edit - does not appear in the list
- Verify no archived content appears in the `Unpublished Content` section (to see a list of archived pages on the backend, go to https://tq-jsonfix-chug.pantheonsite.io/admin/content/moderated?title=&type=All&moderation_state=editorial-archived&langcode=All )
